src/models/simulation.ts

```diff
@@ -18,6 +19,14 @@ const DEFAULT_ZONE_DIVISION = {
   ]
 };
 
+export type Event = "yearChange" | "restart" | "start";
+
+export interface ISimulationSnapshot {
+  time: number;
+  yearlyVegetationStatistics: VegetationStatistics[];
```

Comment:
Note that the yearly vegetation statistic is already an array containing all previous years. We likely don't need to copy this array repeatedly, as it results in O(n^2) size for the snapshots. By year 240, we'll have 240 snapshots, and each snapshot will contain between 240 and 1 actual statistics.

I don't recall how much data we keep in these statistics, but it can grow very quickly.

When going back in time, it's probably sufficient to slice the "master" statistic array to the desired length.